src/hooks/connectWeb3/useActiveWeb3React.tsx

```diff
@@ -163,38 +62,63 @@ function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: strin
   )
 }
 
-function useActiveWalletProvider(): Web3Provider | undefined {
-  return getPriorityConnector(...connections).usePriorityProvider() as Web3Provider
+interface ActiveWeb3ProviderProps {
+  provider?: Eip1193Provider | JsonRpcProvider
+  jsonRpcEndpoint?: string | JsonRpcProvider
+}
+
+export function ActiveWeb3Provider({
+  provider: propsProvider,
+  jsonRpcEndpoint,
+  children,
+}: PropsWithChildren<ActiveWeb3ProviderProps>) {
+  const metaMaskConnection = useMemo(
+    () => toWeb3Connection(initializeConnector<MetaMask>((actions) => new MetaMask({ actions }))),
+    []
+  )
+  const walletConnectConnectionQR = useMemo(() => getWalletConnectConnection(false, jsonRpcEndpoint), [jsonRpcEndpoint]) // WC via tile QR code scan
+  const walletConnectConnectionPopup = useMemo(
+    () => getWalletConnectConnection(true, jsonRpcEndpoint),
+    [jsonRpcEndpoint]
+  ) // WC via built-in popup
+
+  const networkConnection = useMemo(() => {
+    let network: string[] | JsonRpcProvider[]
+    if (JsonRpcProvider.isProvider(jsonRpcEndpoint)) {
+      network = [jsonRpcEndpoint]
+    } else {
+      network = [jsonRpcEndpoint ?? '']
+    }
+    const urlMap = { [SupportedChainId.MAINNET]: network }
+    const connection = toWeb3Connection(initializeConnector<Network>((actions) => new Network({ actions, urlMap })))
+    connection[0].activate() // Activate network immediately after instantiation
```

Comment:
i think you can do this where you try to eagerly connect all the wallets, except try to use activate if eager connection not available